src/Tests/Startup.cs

```diff
@@ -63,6 +63,7 @@ public static void AddMocks(this IServiceCollection services)
             var shell = new MockShell();
             services.AddSingleton<IShellServer>(shell);
             services.AddSingleton<IShellRouter>(new MockShellRouter(shell));
```

Comment:
Is this line still necessary, with porting from shell to comms?

Reply:
It should be, yes. The shell router is still there and used to handle shell messages, including `comm_open` and `comm_msg`. This line makes sure that a mock for the shell router is made available during testing, allowing us to test functionality that depends on Microsoft.Jupyter.Core types without instantiating a real kernel.